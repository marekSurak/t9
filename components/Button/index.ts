import { styled } from 'styles/stitches.config'

export const Button = styled('button', {
  display: 'flex',
  flexFlow: 'column',
  justifyContent: 'center',
  alignItems: 'center',
  background: '$lightGray',

  border: 'none',
  borderRadius: '50%',
  padding: '$3',
  width: '$5',
  height: '$5',

  '&:hover': {
    background: '$dimGray',
  },
})
