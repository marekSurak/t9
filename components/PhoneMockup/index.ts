import { styled } from 'styles/stitches.config'

export const PhoneMockup = styled('div', {
  display: 'flex',
  flexFlow: 'column',
  alignItems: 'center',
  width: '$phoneWidth',
  height: '$phoneHeight',
  overflow: 'hidden',
  padding: '$1 $3',

  border: '1px solid black',
})
