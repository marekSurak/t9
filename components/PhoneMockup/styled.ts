import { styled } from 'styles/stitches.config'

export const PhoneContainer = styled('div', {
  position: 'relative',
})

export const PhoneContent = styled('div', {
  display: 'flex',
  flexFlow: 'column',
  alignItems: 'center',
  justifyContent: 'space-between',
  width: '$phoneWidth',
  height: '$phoneHeight',
  overflow: 'hidden',
  padding: '$4 $5',
})

export const ImageContainer = styled('div', {
  position: 'absolute',
  top: '0',
  zIndex: '$phone',
})
