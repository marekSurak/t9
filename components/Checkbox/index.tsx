import type { FC } from 'react'
import React from 'react'

import { CheckboxContainer } from './styled'

interface IProps {
  checked: boolean
  onClick: () => void
}

export const CHECKBOX_ID = 'checkbox'

const Checkbox: FC<IProps> = ({ checked, onClick }) => (
  <CheckboxContainer>
    <label htmlFor={CHECKBOX_ID}>Use word list</label>
    <input
      id={CHECKBOX_ID}
      type="checkbox"
      checked={checked}
      onChange={onClick}
    />
  </CheckboxContainer>
)

export const MemoizedCheckbox = React.memo(Checkbox)
