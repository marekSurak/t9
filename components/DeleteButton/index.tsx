import type { FC } from 'react'

import { Button, ButtonContainer } from './styled'

interface IProps {
  onClick: () => void
  children: React.ReactNode
}

export const DeleteButton: FC<IProps> = ({ onClick, children }) => (
  <ButtonContainer>
    <Button type="button" onClick={onClick}>
      {children}
    </Button>
  </ButtonContainer>
)
