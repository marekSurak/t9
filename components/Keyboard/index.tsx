import type { FC } from 'react'
import React from 'react'

import { Button } from 'components/Button'
import { keyboardButtons } from 'constants/keyboard'

import { ButtonNumber, ButtonText, KeyboardContainer } from './styled'

interface IProps {
  onClick: (value: number) => void
}

const Keyboard: FC<IProps> = ({ onClick }) => (
  <KeyboardContainer>
    {keyboardButtons.map((button) => (
      <Button
        key={button.value}
        type="button"
        onClick={() => onClick(button.value)}
      >
        <ButtonNumber>{button.value}</ButtonNumber>
        <ButtonText>{button.label}</ButtonText>
      </Button>
    ))}
  </KeyboardContainer>
)

export const MemoizedKeyboard = React.memo(Keyboard)
