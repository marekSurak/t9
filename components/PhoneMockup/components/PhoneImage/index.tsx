import Image from 'next/image'
import React from 'react'

import IphoneImageSrc from 'public/images/iphone.png'

import { ImageContainer } from './styled'

const PhoneImage = () => (
  <ImageContainer>
    <Image
      src={IphoneImageSrc}
      width="372"
      height="750"
      alt="Iphone image"
      placeholder="blur"
    />
  </ImageContainer>
)

export const MemoizedPhoneImage = React.memo(PhoneImage)
