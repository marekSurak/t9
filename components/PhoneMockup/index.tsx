import Image from 'next/image'
import type { FC } from 'react'

import IphoneImageSrc from 'public/images/iphone.png'

import { PhoneContainer, ImageContainer, PhoneContent } from './styled'

interface IProps {
  children: React.ReactNode
}

export const PhoneMockup: FC<IProps> = ({ children }) => (
  <PhoneContainer>
    <ImageContainer>
      <Image
        src={IphoneImageSrc}
        width="372"
        height="750"
        alt="Iphone image"
        placeholder="blur"
      />
    </ImageContainer>
    <PhoneContent>{children}</PhoneContent>
  </PhoneContainer>
)
