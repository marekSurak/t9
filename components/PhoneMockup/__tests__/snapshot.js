import { render } from '@testing-library/react'
import { PhoneMockup } from 'components/PhoneMockup'
import { Keyboard } from 'components/Keyboard'
import { DeleteButton } from 'components/DeleteButton'

it('Renders Phone frame unchanged', () => {
  const { container } = render(
    <PhoneMockup>
      <Keyboard />
      <DeleteButton onClick={() => {}}>x</DeleteButton>
    </PhoneMockup>
  )
  expect(container).toMatchSnapshot()
})
