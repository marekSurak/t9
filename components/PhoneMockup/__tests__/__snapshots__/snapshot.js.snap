// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Phone frame unchanged 1`] = `
<div>
  <div
    class="c-cmpvrW"
  >
    <div
      class="c-iHsKCG"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27372%27%20height=%27750%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Iphone image"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(data:image/png;base64,imagedata);"
        />
        <noscript />
      </span>
    </div>
    <div
      class="c-kiBZOh"
    >
      <section
        class="c-eMcnfs"
      >
        <button
          class="c-diLMKw"
          type="button"
        >
          <span
            class="c-jgiWaE"
          >
            1
          </span>
          <span
            class="c-EDrvg"
          />
        </button>
        <button
          class="c-diLMKw"
          type="button"
        >
          <span
            class="c-jgiWaE"
          >
            2
          </span>
          <span
            class="c-EDrvg"
          >
            ABC
          </span>
        </button>
        <button
          class="c-diLMKw"
          type="button"
        >
          <span
            class="c-jgiWaE"
          >
            3
          </span>
          <span
            class="c-EDrvg"
          >
            DEF
          </span>
        </button>
        <button
          class="c-diLMKw"
          type="button"
        >
          <span
            class="c-jgiWaE"
          >
            4
          </span>
          <span
            class="c-EDrvg"
          >
            GHI
          </span>
        </button>
        <button
          class="c-diLMKw"
          type="button"
        >
          <span
            class="c-jgiWaE"
          >
            5
          </span>
          <span
            class="c-EDrvg"
          >
            JKL
          </span>
        </button>
        <button
          class="c-diLMKw"
          type="button"
        >
          <span
            class="c-jgiWaE"
          >
            6
          </span>
          <span
            class="c-EDrvg"
          >
            MNO
          </span>
        </button>
        <button
          class="c-diLMKw"
          type="button"
        >
          <span
            class="c-jgiWaE"
          >
            7
          </span>
          <span
            class="c-EDrvg"
          >
            PQRS
          </span>
        </button>
        <button
          class="c-diLMKw"
          type="button"
        >
          <span
            class="c-jgiWaE"
          >
            8
          </span>
          <span
            class="c-EDrvg"
          >
            TUV
          </span>
        </button>
        <button
          class="c-diLMKw"
          type="button"
        >
          <span
            class="c-jgiWaE"
          >
            9
          </span>
          <span
            class="c-EDrvg"
          >
            WXYZ
          </span>
        </button>
        <button
          class="c-diLMKw"
          type="button"
        >
          <span
            class="c-jgiWaE"
          >
            0
          </span>
          <span
            class="c-EDrvg"
          >
            +
          </span>
        </button>
      </section>
    </div>
  </div>
</div>
`;
